File: leet/c/553_optimal_division.c
#include <stdlib.h>
#include <stdio.h>

char* optimalDivision(int* nums, int numsSize) {
	char *exp = malloc(numsSize * 10 * 2);
	int i;

	if (numsSize == 1) {
		sprintf(exp, "%d", nums[0]);
	} else if (numsSize == 2) {
		sprintf(exp, "%d/%d", nums[0], nums[1]);
	} else {
		sprintf(exp, "%d/(%d", nums[0], nums[1]);
		for (i = 2; i < numsSize; i++)
			sprintf(exp, "%s/%d", exp, nums[i]);
		sprintf(exp, "%s)", exp);
	}
	return exp;
}

int main()
{
		int nums[] = {1000, 100, 10, 2};
		printf("%s\n", optimalDivision(nums, 4));
}

File: leet/python/553_optimal_division.py
class Solution(object):
    def optimalDivision(self, nums):
        """
        :type nums: List[int]
        :rtype: str
        """
        nums = [str(num) for num in nums]
        if len(nums) == 1:
           return nums[0]
        elif len(nums) == 2:
           return nums[0] + '/' + nums[1]

        nums[1] = '(' + nums[1]
        nums[len(nums) - 1] += ')'
        return '/'.join(nums)
